src/Core/Compiler/CompilerService.cs

```diff
@@ -248,54 +245,52 @@ public IEnumerable<QsNamespaceElement> IdentifyElements(string source)
                     options);
 
                 // Generate the assembly from the C# compilation:
-                using (var ms = new MemoryStream())
-                using (var bsonStream = new MemoryStream())
-                {
-                    using var writer = new BsonDataWriter(bsonStream) { CloseOutput = false };
-                    var fromSources = qsCompilation.Namespaces.Select(ns => FilterBySourceFile.Apply(ns, s => s.EndsWith(".qs")));
-                    Json.Serializer.Serialize(writer, new QsCompilation(fromSources.ToImmutableArray(), qsCompilation.EntryPoints));
+                var fromSources = qsCompilation.Namespaces.Select(ns => FilterBySourceFile.Apply(ns, s => s.EndsWith(".qs")));
+                var syntaxTree = new QsCompilation(fromSources.ToImmutableArray(), qsCompilation.EntryPoints);
```

Comment:
Apologies for the ugly-looking diff here. It's mostly whitespace/indentation changes, since I moved to inline `using` statements. Choosing "Hide whitespace changes" should make it easier to review.